### PR TITLE
Determine user agent version programmatically

### LINKED ISSUE #294: Determine version for User-Agent programatically
The version number is hardcoded in both the production code and the test. This should be determined programmatically, most likely from some kind of generated code or properties file.

### PR DESCRIPTION
**Pull request by max-vogler**
### Summary

Loading the user agent version using a Maven-built resource file during runtime and by parsing `pom.xml` during tests.

Fixes #294

### CODE REVIEW

File: src/main/java/com/ibm/watson/developer_cloud/util/RequestUtils.java [hunk 1]
```diff
@@ -159,25 +163,35 @@ public static String getUserAgent() {
...
   }
 
+  private static String loadSdkVersion() {
+    InputStream inputStream = RequestUtils.class.getClassLoader().getResourceAsStream("version.properties");
+    Properties properties = new Properties();
+
+    try {
+      properties.load(inputStream);
+    } catch (IOException e) {
+      e.printStackTrace();
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/developer_cloud/util/RequestUtils.java]: Codacy Issue found: [Avoid printStackTrace(); use a logger call instead.](https://www.codacy.com/app/watson-developer-cloud/java-sdk/file/2730629156/issues/source?bid=3303699&fileBranchId=3349920#l173)

File: src/main/java/com/ibm/watson/developer_cloud/util/RequestUtils.java [hunk 2]
```diff
@@ -159,25 +163,35 @@ public static String getUserAgent() {
     return userAgent;
   }
 
+  private static String loadSdkVersion() {
+    InputStream inputStream = RequestUtils.class.getClassLoader().getResourceAsStream("version.properties");
```
Comment (germanattanasio) [inline src/main/java/com/ibm/watson/developer_cloud/util/RequestUtils.java]: Codacy Issue found: [In J2EE, getClassLoader() might not work as expected.  Use Thread.currentThread().getContextClassLoader() instead.](https://www.codacy.com/app/watson-developer-cloud/java-sdk/file/2730629156/issues/source?bid=3303699&fileBranchId=3349920#l167)

**Comment by germanattanasio** [comment]
Look at the codacy comments

**Comment by DimuthRuwantha** [comment]
returns "Could not load version.properties" in built war file

**Pull request merged**